libr/util/file.c

```diff
@@ -360,7 +360,11 @@ R_API char *r_file_slurp(const char *str, R_NULLABLE size_t *usz) {
 				buf = nbuf;
 				size_t r = fread (buf + size, 1, BS, fd);
 				size += r;
-			} while (!feof (fd));
+			} while (!feof (fd) && !ferror (fd));
```

Comment:
call ferror once inside the loop and break or return NULL directly? in there before incrementing the size? usz doesnt needs to be set if the read fails